### DISCUSSION TITLE
Error while loading latest version of repos on vscode

### DISCUSSION THREAD

**Discussion by digrapsas** [fortran-lang/fortls#367] [Q&A]
Hi there, I'd like to give it a go to write an LSP response for the folding range. I followed the instructions in contributing.md and managed to update the fortls executable without any issue. When I open a *.f90 file though with vscode, it fails to load fortls, the error message is the following:

![screen](https://github.com/fortran-lang/fortls/assets/88543231/5093a2fe-2d37-4f45-b949-72a19e0ef573)

I also have some unit tests failing, I don't know if this is somehow related...

![screen2](https://github.com/fortran-lang/fortls/assets/88543231/29514c11-8e64-4c59-bd3b-122ef438e610)

**Comment by gnikit** [comment]
Vscode probably does not work because you are not giving it a valid path to the fortls binary or in your version of fortls you have broken something. It's hard to tell without looking at settings and or your code.

The specific test requires network access and it's supposed to check that the auto-update mechanism works. There might be other reasons why it fails, but I would not fixate on that.

On a side note, the task you have undertaken is non-trivial and probably needs regular guidance on my behalf. Which is why it's currently part of a proposed 2-3 month GSoC project. 
It's a lot of work and I'm not sure I will have time to supervise. 

That being said if you want to give it a go I would start by implementing a response lsp function first, followed by a debug interface so that you can use the cli debug interface instead of vscode to test your code. I would also start backwards i.e. write a very simple test and then write the code to make it work.

**Reply by digrapsas** [reply]
Fortls is working fine with the binary downloaded with pip. Then I download and build the fortls repository as described in CONTRIBUTING.md, without any modifications at all. The error mentioned above arises... I have gone back and forth a couple of times with the same results.

I see this development as an opportunity to check something new. This kind of project is not in my pack of skills, but from time to time I like exploring new things. So you don't have to worry at all, the development won't be intense. In fact, I cannot guarantee at all that the project will go very far, for the moment I'm merely checking if I can do anything at all.

I'll have a look on lsp functions and the cli interface, thanks.

**Reply by digrapsas** [reply]
@gnikit I managed to set up an environment for debugging and I pushed a first version of the foldingRange provider. It is very limited for the time being, mainly because it is only adjusted to "if" and "program" blocks, but I think the essential is there and the rest of the work would consist in adding the rest of the blocks and testing to more complex cases. Let me know what you think and if you'd like to take this any further. 

From my side I'm hesitating between:
1/ keeping the current strategy for assigning the FoldingRange[] and 
2/ adding endlines to the existing blocks (e.g. "If" block doesn't provide any) and creating the missing blocks with endlines.

The second option is by more delicate and time consuming, and I'm unsure that it would provide any benefit for the foldingRange (other than more readable code), unless the developed features will be used elsewhere.
